File: backend/src/main/java/es/udc/paproject/backend/model/services/SportTestService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Province;
import es.udc.paproject.backend.model.entities.SportTest;
import es.udc.paproject.backend.model.entities.SportTestType;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;

import java.time.LocalDate;
import java.util.List;

public interface SportTestService {

    List<Province> findAllProvinces();

    List<SportTestType> findAllSportTestTypes();

    SportTest findSportTestById(Long id) throws InstanceNotFoundException;

    Block<SportTest> findSportTests(Long provinceId, Long typeId, LocalDate startDate, LocalDate finishDate, int page, int size);

}

File: frontend/src/backend/sportTestService.js
import { config, appFetch } from './appFetch'

const baseUrl = '/sportTestsSearch'

export const findSportTests = (params, onSuccess) => {
  const { startDate, finishDate, provinceId, typeId, page } = params
  let path = `${baseUrl}/sportTests?page=${page}`

  path += startDate ? `&startDate=${startDate}` : ''
  path += finishDate ? `&finishDate=${finishDate}` : ''
  path += provinceId ? `&provinceId=${provinceId}` : ''
  path += typeId ? `&typeId=${typeId}` : ''

  appFetch(path, config('GET'), onSuccess)
}

export const findAllProvinces = onSuccess =>
  appFetch(`${baseUrl}/provinces`, config('GET'), onSuccess)

export const findAllSportTestTypes = onSuccess =>
  appFetch(`${baseUrl}/types`, config('GET'), onSuccess)

export const findSportTestById = (id, onSuccess) =>
  appFetch(`${baseUrl}/sportTests/${id}`, config('GET'), onSuccess)
